src/DesktopPixmap.hpp

```diff
@@ -0,0 +1,54 @@
+#ifndef DESKTOP_PIXMAP_HPP
+#define DESKTOP_PIXMAP_HPP
+
+#include <string>
+#include <xcb/xproto.h>
+
+enum PixmapFormat
+{
+  kXyPixmap = 1,
+  kZPixmap = 2
+};
+
+/**
+ * Captures and stores desktop screenshot in the specified pixmap format
+ *
+ * XXX Maybe use static xcb_connection_t?
+ */
+class DesktopPixmap
+{
+public:
+  inline static xcb_gcontext_t gc_; ///< Graphic context
+  short x;                          ///< TODO Might not need this
+  short y;                          ///< TODO Might not need this
+  uint16_t width;
+  uint16_t height;
+  uint8_t *image_ptr;     ///< Pointer to the image structure. TODO Make smart
+  uint32_t length;        ///< Length of the XImage data structure
+  std::string name;       ///< _NET_WM_NAME of display
+  PixmapFormat format;    /* XY_PIXMAP = 1, Z_PIXMAP = 2 */
+  xcb_pixmap_t pixmap_id; ///< Constant id for the screenshot's pixmap
+
+  DesktopPixmap (short x,         ///< x coordinate of the top left corner
+                 short y,         ///< y coordinate of the top left corner
+                 uint16_t height, ///< Height of the display
+                 uint16_t width,  ///< Width of the display
+                 const std::string &name ///< Name of the display (_NET_WM_NAME)
+  );
+  ~DesktopPixmap ();
+
+  /**
+   * TODO Rename to something more fitting
+   */
+  void saveImage (xcb_connection_t *c, xcb_screen_t &screen,
+                  PixmapFormat format = kZPixmap // Default to more efficient
+  );
+  void putImage (xcb_connection_t *c, xcb_screen_t &screen);
+
+private:
+  // XXX Do I need static here?
```

Comment:
static methods are used to use class without initializing the object of the class. You can create gc without using this class at all, so not really, I guess. Not quite sure though, tbh.

Reply:
https://stackoverflow.com/a/6446025
Yes, declaring this function as a class member is redundant, but it looks good, so why not